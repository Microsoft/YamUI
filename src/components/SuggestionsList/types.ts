/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */

import { AvatarProps } from '../Avatar';

export enum SuggestionsListItemTemplate {
  TEXT = 'text',
  AVATAR_ROUND = 'mediaRound',
  MEDIA_SOFT = 'mediaSoft',
  MEDIA_SQUARE = 'mediaSquare',
}

export type CustomizableAvatarProps = 'badgeContent' | 'badgeDescription';

export interface BaseSuggestionItem {
  id: string | number;
  name: string;
}
export interface TextSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.TEXT;
}
export interface MediaSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.MEDIA_SOFT | SuggestionsListItemTemplate.MEDIA_SQUARE;
  imageUrl?: string;
  description?: string;
}
export interface AvatarSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.AVATAR_ROUND;
  imageUrl?: string;
  avatarProps?: Pick<AvatarProps, CustomizableAvatarProps>;
  description?: string;
}
export type SuggestionItem = TextSuggestionItem | MediaSuggestionItem | AvatarSuggestionItem;
