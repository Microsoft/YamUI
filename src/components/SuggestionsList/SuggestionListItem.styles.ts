/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { mergeStyleSets } from '@uifabric/styling';
import { memoizeFunction } from '@uifabric/utilities';
import { getTheme } from '../../util/colors';
import { fontWeights } from '../../util/styles/fonts';

export interface SuggestionsListItemStyleProps {
  isSelected: boolean;
}

const getMemoizedClassNames = memoizeFunction((styleProps, theme) => {
  const { isSelected } = styleProps;

  return mergeStyleSets({
    root: {
      cursor: 'pointer',
      padding: '0.4rem 1.2rem 0.8rem 1.2rem',
      backgroundColor: isSelected ? theme.palette.neutralLight : undefined,
      selectors: {
        ':hover': {
          backgroundColor: theme.palette.neutralLighter,
        },
      },
    },
    highlight: {
      fontWeight: fontWeights.bold,
    },
  });
});

export const getClassNames = (styleProps: SuggestionsListItemStyleProps) => {
  const theme = getTheme();
  return getMemoizedClassNames(styleProps, theme);
};
