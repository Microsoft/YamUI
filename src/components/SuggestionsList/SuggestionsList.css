/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
@import "src/css/variables/colors.css";

.y-suggestions-list--with-status .y-suggestions-list--loading {
  height: 5.2rem;
}

.y-suggestions-list--with-status.y-suggestions-list--with-results .y-suggestions-list--loading {
  height: 4.7rem;
}

.y-suggestions-list--results {
  padding: 0.4rem 0;
}

.y-suggestions-list--with-status .y-suggestions-list--results {
  padding: 0.4rem 0 0 0;
  border-bottom: 0.1rem solid $borderColor__tertiary;
}
