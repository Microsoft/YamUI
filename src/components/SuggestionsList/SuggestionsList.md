### Examples

With no results:

```js { "props": { "data-example": "No Results" } }
const CalloutTrigger = require('../../demos/CalloutTrigger');

<div style={{ minHeight: '50px' }}>
  <SuggestionsList noResultsText="No results found" />
</div>
```

When loading:

```js { "props": { "data-example": "When Loading" } }
const CalloutTrigger = require('../../demos/CalloutTrigger');

<div style={{ minHeight: '50px' }}>
  <SuggestionsList
    isLoading={true}
    loadingText={'Searching for \'Cris\''}
  />
</div>
```

With results:

```js { "props": { "data-example": "When Results" } }
const CalloutTrigger = require('../../demos/CalloutTrigger');

<div style={{ minHeight: '250px' }}>
  <SuggestionsList
    searchText="Cris"
    loadingText={'Searching for \'Cris\''}
    onItemSelected={() => {}}
    groupedItems={[
      {
        title: 'Suggested Users',
        items: [{
          id: 1,
          imageUrl: '',
          name: 'Cristiana Cline',
          description: 'Data Analyst'
        }, {
          id: 2,
          imageUrl: '',
          name: 'Cristian Sayers',
          description: 'Front-End Developer'
        }, {
          id: 3,
          imageUrl: '',
          name: 'Johanna Cristian Lorenz',
          description: 'Information Architecture'
        }, {
          id: 4,
          imageUrl: '',
          name: 'Lidia Holloway Cristina',
          description: 'Senior Designer'
        }, {
          id: 5,
          imageUrl: '',
          name: 'Nestor Cristian Wilke',
          description: 'Product Manager'
        }]
      }
    ]}
    selectedId={2}
  />
</div>
```

With results, when loading:

```js { "props": { "data-example": "When Results" } }
const CalloutTrigger = require('../../demos/CalloutTrigger');

<div style={{ minHeight: '295px', position: 'relative' }}>
  <SuggestionsList
    left={0}
    top={0}
    isLoading={true}
    searchText="Cris"
    loadingText={'Searching for \'Cris\''}
    onItemSelected={() => {}}
    groupedItems={[
      {
        title: 'Suggested Users',
        items: [{
          id: 1,
          imageUrl: '',
          name: 'Cristiana Cline',
          description: 'Data Analyst'
        }, {
          id: 2,
          imageUrl: '',
          name: 'Cristian Sayers',
          description: 'Front-End Developer'
        }, {
          id: 3,
          imageUrl: '',
          name: 'Johanna Cristian Lorenz',
          description: 'Information Architecture'
        }, {
          id: 4,
          imageUrl: '',
          name: 'Lidia Holloway Cristina',
          description: 'Senior Designer'
        }, {
          id: 5,
          imageUrl: '',
          name: 'Nestor Cristian Wilke',
          description: 'Product Manager'
        }]
      }
    ]}
    selectedId={2}
  />
</div>
```

```js { "props": { "data-example": "Right alignment" } }
const CalloutTrigger = require('../../demos/CalloutTrigger');

<div style={{ minHeight: '295px', position: 'relative' }}>
  <SuggestionsList
    right={0}
    top={0}
    isLoading={true}
    searchText="Cris"
    loadingText={'Searching for \'Cris\''}
    onItemSelected={() => {}}
    groupedItems={[
      {
        title: 'Suggested Users',
        items: [{
          id: 1,
          imageUrl: '',
          name: 'Cristiana Cline',
          description: 'Data Analyst'
        }, {
          id: 2,
          imageUrl: '',
          name: 'Cristian Sayers',
          description: 'Front-End Developer'
        }, {
          id: 3,
          imageUrl: '',
          name: 'Johanna Cristian Lorenz',
          description: 'Information Architecture'
        }, {
          id: 4,
          imageUrl: '',
          name: 'Lidia Holloway Cristina',
          description: 'Senior Designer'
        }, {
          id: 5,
          imageUrl: '',
          name: 'Nestor Cristian Wilke',
          description: 'Product Manager'
        }]
      }
    ]}
    selectedId={2}
  />
</div>
```

