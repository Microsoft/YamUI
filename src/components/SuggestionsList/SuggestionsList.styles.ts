/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { mergeStyleSets, ITheme } from '@uifabric/styling';
import { memoizeFunction } from '@uifabric/utilities';
import { getGutterValue } from '../../util/styles/gutters';
import { borderStylePopOver } from '../../util/styles/borders';

export interface SuggestionsClassNameProps {
  isLoading?: boolean;
  hasResults: boolean;
  theme: ITheme;
}

export interface SuggestionsListClassNames {
  root: string;
  spinnerWrapper: string;
  noResultsWrapper: string;
}

export const getClassNames = memoizeFunction((classNameProps: SuggestionsClassNameProps) => {
  const { isLoading, hasResults, theme } = classNameProps;

  return mergeStyleSets({
    root: {
      ...borderStylePopOver,
      width: '280px',
      paddingTop: hasResults ? getGutterValue(1) : undefined,
      paddingBottom: hasResults && !isLoading ? getGutterValue(1) : undefined,
    },
    spinnerWrapper: {
      paddingBottom: '14px',
      paddingTop: hasResults ? '13px' : '14px',
      borderTop: hasResults ? `1px solid ${theme.palette.neutralLighter}` : undefined,
    },
    noResultsWrapper: {
      paddingTop: '15px',
      paddingBottom: '17px',
    },
  });
});
