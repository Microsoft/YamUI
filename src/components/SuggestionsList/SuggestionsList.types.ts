/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { AvatarProps } from '../Avatar';
import { BaseComponentProps } from '../../util/BaseComponent/props';

export enum SuggestionsListItemTemplate {
  TEXT = 'text',
  AVATAR_ROUND = 'mediaRound',
  MEDIA_SOFT = 'mediaSoft',
  MEDIA_SQUARE = 'mediaSquare',
}

export type CustomizableAvatarProps = 'badgeContent' | 'badgeDescription';

export interface BaseSuggestionItem {
  id: string | number;
  name: string;
}

export interface TextSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.TEXT;
}

export interface MediaSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.MEDIA_SOFT | SuggestionsListItemTemplate.MEDIA_SQUARE;
  imageUrl?: string;
  description?: string;
}

export interface AvatarSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.AVATAR_ROUND;
  imageUrl?: string;
  avatarProps?: Pick<AvatarProps, CustomizableAvatarProps>;
  description?: string;
}

export type SuggestionItem = TextSuggestionItem | MediaSuggestionItem | AvatarSuggestionItem;

export interface SuggestionItemGroup {
  title: string;
  items: SuggestionItem[];
}

export interface SuggestionsListProps extends BaseComponentProps {
  /**
   * The groups of items to render.
   */
  groupedItems: SuggestionItemGroup[];

  /**
   * The active search that produced the current state.
   */
  searchText: string;

  /**
   * Text to show next to the loading spinner.
   */
  loadingText: string;

  /**
   * Text to show when the results are empty for the active search.
   */
  noResultsText: string;

  /**
   * Flag to indicate that a search is in flight
   * @default false
   */
  isLoading?: boolean;

  /**
   * The index of current keyboard selection.
   */
  selectedId?: string | number;

  /**
   * Called when an item in a group is selected by clicking.
   */
  onItemSelected(id: string | number): void;
}

export interface SuggestionsListItemProps extends BaseComponentProps {
  item: SuggestionItem;
  searchText: string;
  isSelected: boolean;
  onSelect(id: SuggestionItem['id']): void;
}
