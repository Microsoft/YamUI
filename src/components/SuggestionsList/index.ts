/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export { default } from './SuggestionsList';
export * from './SuggestionsList';
export { default as SuggestionsListItem } from './SuggestionsListItem';
export * from './SuggestionsListItem';
