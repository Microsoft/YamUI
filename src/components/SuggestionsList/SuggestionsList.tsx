/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import SuggestionsListItem, { SuggestionItem } from './SuggestionsListItem';
import Spinner from '../Spinner';
import Text, { TextSize, TextColor } from '../Text';
import Callout, { DirectionalHint } from '../Callout';
import Block, { GutterSize } from '../Block';
import './SuggestionsList.css';
import { IPoint } from 'office-ui-fabric-react/lib/utilities/positioning';

export interface SuggestionsListProps extends NestableBaseComponentProps {
  /**
   * The target that the Callout should try to position itself based on.
   */
  target: HTMLElement | IPoint;
  /**
   * The active search that produced the current state.
   */
  searchText: string;
  /**
   * Text to show next to the loading spinner.
   */
  loadingText: string;
  /**
   * Text to show when the results are empty for the active search.
   */
  noResultsText: string;
  /**
   * Flag to indicate that a search is in flight
   * @default false
   */
  isLoading?: boolean;
  /**
   * The groups of items to render.
   */
  groupedItems?: SuggestionItemGroupProps[];
  /**
   * The index of current keyboard selection.
   */
  selectedId?: string | number;
  /**
   * Called when an item in a group is selected by clicking.
   */
  onItemSelected(id: string | number): void;
}

interface SuggestionsListWithResultsProps extends SuggestionsListProps {
  groupedItems: SuggestionItemGroupProps[];
}

export interface SuggestionItemGroupProps {
  title: string;
  items: SuggestionItem[];
}

const hasResults = (props: SuggestionsListProps): props is SuggestionsListWithResultsProps => {
  return !!props.groupedItems && props.groupedItems.length > 0;
};

const withResultsClass = 'y-suggestions-list--with-results';
const withStatusClass = 'y-suggestions-list--with-status';

/**
 * A `SuggestionsList` displays a list of search results in a dropdown.
 */
export default class SuggestionsList extends React.PureComponent<SuggestionsListProps> {
  public render() {
    const status = this.getSearchStatus();
    const results = this.getGroupedResults();

    const classNames = ['y-suggestions-list'];
    if (status) {
      classNames.push(withStatusClass);
    }
    if (results) {
      classNames.push(withResultsClass);
    }

    return (
      <Callout
        calloutWidth={280}
        doNotLayer={true}
        directionalHint={DirectionalHint.bottomLeftEdge}
        directionalHintFixed={true}
        target={this.props.target}
        className={classNames.join(' ')}
        isBeakVisible={false}
        minPagePadding={-500} /* allow to run past bottom of viewport */
      >
        {results}
        {status}
      </Callout>
    );
  }

  private getGroupedResults() {
    if (!hasResults(this.props)) {
      return null;
    }
    return (
      <ul className="y-suggestions-list--results">
        {this.props.groupedItems.map(this.getGroupResults)}
      </ul>
    );
  }

  private getGroupResults = (group: SuggestionItemGroupProps) => {
    const items = group.items.map(this.getResultItem);
    return (
      <li key={group.title}>
        <ul>{items}</ul>
      </li>
    );
  };

  private getResultItem = (item: SuggestionItem) => {
    const { searchText, selectedId, onItemSelected } = this.props;
    const isSelected = item.id === selectedId;
    return (
      <li key={item.id}>
        <SuggestionsListItem
          searchText={searchText}
          isSelected={isSelected}
          onSelect={onItemSelected}
          {...item}
        />
      </li>
    );
  };

  private getSearchStatus() {
    return this.props.isLoading ? this.getLoading() : this.getNoResults();
  }

  private getLoading() {
    return (
      <Spinner
        text={this.props.loadingText}
        className="y-suggestions-list--loading"
        isCentered={true}
      />
    );
  }

  private getNoResults() {
    if (hasResults(this.props)) {
      return null;
    }
    return (
      <Block textAlign="center" padding={GutterSize.LARGE} textSize={TextSize.SMALL}>
        <Text color={TextColor.METADATA}>{this.props.noResultsText}</Text>
      </Block>
    );
  }
}
