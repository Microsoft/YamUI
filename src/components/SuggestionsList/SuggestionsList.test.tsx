/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, mount, ShallowWrapper, ReactWrapper } from 'enzyme';
import CustomizableSuggestionsList, { SuggestionsList } from './SuggestionsList';
import { SuggestionsListProps, SuggestionsListItemTemplate } from './SuggestionsList.types';
import Customizer, { defaultTheme } from '../Customizer';

describe('<SuggestionsList />', () => {
  let component: ShallowWrapper<SuggestionsListProps>;
  let onItemSelected: jest.Mock;
  let options: Partial<SuggestionsListProps>;

  const getProps = (overrides: Partial<SuggestionsListProps>) => {
    onItemSelected = jest.fn();

    const defaultProps = {
      onItemSelected,
      searchText: 'searchText',
      loadingText: 'loadingText',
      noResultsText: 'noResultsText',
      isLoading: false,
      selectedId: 2,
      groupedItems: [],
    };

    return {
      ...defaultProps,
      ...overrides,
    };
  };

  beforeEach(() => {
    options = {};
  });

  describe('when isLoading=true', () => {
    beforeEach(() => {
      options = {
        ...options,
        isLoading: true,
      };
    });

    describe('without results', () => {
      beforeEach(() => {
        component = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with results', () => {
      beforeEach(() => {
        options = {
          ...options,
          groupedItems: [
            {
              title: 'groupTitle',
              items: [
                {
                  template: SuggestionsListItemTemplate.TEXT,
                  id: 'id',
                  name: 'name',
                },
              ],
            },
          ],
        };
        component = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('when isLoading=false', () => {
    beforeEach(() => {
      options = {
        ...options,
        isLoading: false,
      };
    });

    describe('without results', () => {
      beforeEach(() => {
        component = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with results', () => {
      beforeEach(() => {
        options = {
          ...options,
          groupedItems: [
            {
              title: 'groupTitle',
              items: [
                {
                  template: SuggestionsListItemTemplate.TEXT,
                  id: 'id',
                  name: 'name',
                },
              ],
            },
          ],
        };
        component = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('with customizer', () => {
    let mountedComponent: ReactWrapper;
    const theme = defaultTheme;

    beforeEach(() => {
      mountedComponent = mount(
        <Customizer settings={{ theme }}>
          <CustomizableSuggestionsList {...getProps({})} />
        </Customizer>,
      );
    });

    it('receives custom theme', () => {
      expect(mountedComponent.find('CustomizableSuggestionsList').prop('theme')).toBe(theme);
    });
  });
});
