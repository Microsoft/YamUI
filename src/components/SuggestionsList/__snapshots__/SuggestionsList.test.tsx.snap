// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestionsList /> when isLoading=false with results renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-results"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <ul
    className="y-suggestions-list--results"
  >
    <li>
      <ul>
        <li>
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </li>
      </ul>
    </li>
  </ul>
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=false with results when onHover is called updates the state 1`] = `null`;

exports[`<SuggestionsList /> when isLoading=false with results with state.hoveredId=id renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-results"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <ul
    className="y-suggestions-list--results"
  >
    <li>
      <ul>
        <li>
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </li>
      </ul>
    </li>
  </ul>
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=false with results with state.hoveredId=id when the mouse leaves renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-results"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <ul
    className="y-suggestions-list--results"
  >
    <li>
      <ul>
        <li>
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </li>
      </ul>
    </li>
  </ul>
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=false without results renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-status"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <Block
    padding="large"
    textAlign="center"
    textSize="small"
  >
    <Text
      color="metadata"
    >
      noResultsText
    </Text>
  </Block>
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=true with results renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-status y-suggestions-list--with-results"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <ul
    className="y-suggestions-list--results"
  >
    <li>
      <ul>
        <li>
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </li>
      </ul>
    </li>
  </ul>
  <Spinner
    className="y-suggestions-list--loading"
    color="light"
    hideText={false}
    size="medium"
    text="loadingText"
  />
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=true with results when onHover is called updates the state 1`] = `null`;

exports[`<SuggestionsList /> when isLoading=true with results with state.hoveredId=id renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-status y-suggestions-list--with-results"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <ul
    className="y-suggestions-list--results"
  >
    <li>
      <ul>
        <li>
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </li>
      </ul>
    </li>
  </ul>
  <Spinner
    className="y-suggestions-list--loading"
    color="light"
    hideText={false}
    size="medium"
    text="loadingText"
  />
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=true with results with state.hoveredId=id when the mouse leaves renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-status y-suggestions-list--with-results"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <ul
    className="y-suggestions-list--results"
  >
    <li>
      <ul>
        <li>
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </li>
      </ul>
    </li>
  </ul>
  <Spinner
    className="y-suggestions-list--loading"
    color="light"
    hideText={false}
    size="medium"
    text="loadingText"
  />
</Callout>
`;

exports[`<SuggestionsList /> when isLoading=true without results renders as expected 1`] = `
<Callout
  calloutWidth={280}
  className="y-suggestions-list--with-status"
  directionalHint={4}
  directionalHintFixed={true}
  doNotLayer={true}
  isBeakVisible={false}
  target={<div />}
>
  <Spinner
    className="y-suggestions-list--loading"
    color="light"
    hideText={false}
    size="medium"
    text="loadingText"
  />
</Callout>
`;
